src/renderer/components/+workloads-pods/pod-container-env/env-from.tsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import { withInjectables } from "@ogre-tools/injectable-react";
+import { autorun } from "mobx";
+import React, { useEffect } from "react";
+import type { ConfigMapStore } from "../../+config-maps/store";
+import configMapStoreInjectable from "../../+config-maps/store.injectable";
+import { SecretKey } from "../../+config-secrets/secret-key";
+import type { SecretStore } from "../../+config-secrets/store";
+import secretStoreInjectable from "../../+config-secrets/store.injectable";
+import type { EnvFromSource } from "../../../../common/k8s-api/endpoints";
+import type { Logger } from "../../../../common/logger";
+import loggerInjectable from "../../../../common/logger.injectable";
+import { iter, object } from "../../../utils";
+
+export interface ContainerEnvFromSourceProps {
+  envFrom: EnvFromSource[];
+  namespace: string;
+}
+
+interface Dependencies {
+  secretStore: SecretStore;
+  configMapStore: ConfigMapStore;
+  logger: Logger;
+}
+
+const NonInjectedContainerEnvFromSource = ({
+  envFrom,
+  namespace,
+  configMapStore,
+  secretStore,
+  logger,
+}: ContainerEnvFromSourceProps & Dependencies) => {
+  useEffect(() => autorun(() => {
+    for (const { configMapRef, secretRef } of envFrom) {
+      if (secretRef?.name) {
+        secretStore
+          .load({ name: secretRef.name, namespace })
+          .catch(error => logger.warn(`[CONTAINER-ENV]: failed to load Secret ${secretRef.name} in ns=${namespace}`, error));
+      }
+
+      if (configMapRef?.name) {
+        configMapStore
+          .load({ name: configMapRef.name, namespace })
+          .catch(error => logger.warn(`[CONTAINER-ENV]: failed to load ConfigMap ${configMapRef.name} in ns=${namespace}`, error));
+      }
+    }
+  }), []);
```

Comment:
Render time side-effect. Also looks some-what duplicated. Could be wrong. But as said in other comment, I know you are just extracting this as of component, so no changes needed for now.

Reply:
Probably could just replace this with a top level subscribe instead.